Thinktecture.Relay.Server/Repository/UserRepository.cs

```diff
@@ -180,5 +204,63 @@ public bool IsUserNameAvailable(string userName)
 				return user == null;
 			}
 		}
+
+		private DateTime? LockedOutUntil(int? failedAttempts, DateTime? lastFailedAttempt)
+		{
+			if (!failedAttempts.HasValue)
+				return null;
+
+			if (failedAttempts < _configuration.MaxFailedLoginAttempts)
+				return null;
+
+			if (!lastFailedAttempt.HasValue)
+				return null;
+
+			if (lastFailedAttempt + _configuration.FailedLoginLockoutPeriod < DateTime.UtcNow)
+				return null;
+
+			return lastFailedAttempt + _configuration.FailedLoginLockoutPeriod;
+		}
+
+		private bool IsUserLockedOut(DbUser user)
+		{
+			var lockedOutUntil = LockedOutUntil(user.FailedLoginAttempts, user.LastFailedLoginAttempt);
+			if (!lockedOutUntil.HasValue)
+				return false;
+
+			_logger?.Information("User {UserName} is locked out til {LockoutEnd} due to {FailedLoginAttempts} failed login attempts since {LastFailedLoginAttempt}",
```

Comment:
The combination of `since` and the `last attempt` is not quite right. Shouldn't it be something like `User {UserName} is locked out 'til {LockoutEnd} due to {FailedLoginAttempts} failed login attempts (last try was at {LastFailedLoginAttempt})`?